akka-http-core/src/main/scala/akka/http/engine/parsing/ParserSettings.scala

```diff
@@ -66,9 +67,25 @@ object ParserSettings extends SettingsCompanion[ParserSettings]("akka.http.parsi
       c getIntBytes "max-chunk-size",
       Uri.ParsingMode(c getString "uri-parsing-mode"),
       c getBoolean "illegal-header-warnings",
+      ErrorLoggingVerbosity(c getString "error-logging-verbosity"),
       cacheConfig.entrySet.asScala.map(kvp ⇒ kvp.getKey -> cacheConfig.getInt(kvp.getKey))(collection.breakOut),
       _ ⇒ None,
       _ ⇒ None)
   }
+
+  sealed trait ErrorLoggingVerbosity
+  object ErrorLoggingVerbosity {
+    case object Off extends ErrorLoggingVerbosity
+    case object Simple extends ErrorLoggingVerbosity
+    case object Full extends ErrorLoggingVerbosity
+
+    def apply(string: String): ErrorLoggingVerbosity =
+      string match {
+        case "off"    ⇒ Off
+        case "simple" ⇒ Simple
+        case "full"   ⇒ Full
+        case _        ⇒ throw new IllegalArgumentException(string + " is not a legal `error-logging-verbosity` setting")
```

Comment:
We try to enclose values in `[ ]` in exception and log messages (we are not always consistent with this).
Here it would make sense, since it would make it more easy to spot empty string value.